src/sfctl/custom_app.py

```diff
@@ -12,6 +12,112 @@
 import sys
 import shutil
 from knack.util import CLIError
+from msrest.pipeline import ClientRawResponse
+
+# We are disabling some W0212 lint warnings in the following function
+# because of a problem with the generated SDK that does not allow this
+# function to be called upon from within the generated SDK.
+def provision_application_type(client, #pylint: disable=too-many-locals,invalid-name,too-many-arguments
+                               image_store_provision=False,
+                               external_store_provision=False,
+                               async_operation=False,
+                               application_type_build_path=None,
+                               application_package_download_uri=None,
+                               application_type_name=None,
+                               application_type_version=None,
+                               timeout=60, custom_headers=None, raw=False):
+    """Provisions or registers a Service Fabric application type with the
+        cluster using the .sfpkg package in the external store or using the
+        application package in the image store.
+    """
+
+    from azure.servicefabric.models.provision_application_type_description \
+       import (ProvisionApplicationTypeDescription)
+    from azure.servicefabric.models.external_store_provision_application_type_description import (ExternalStoreProvisionApplicationTypeDescription) # pylint: disable=line-too-long
+
+    from azure.servicefabric import models
```

Comment:
Import only the models you use.

Reply:
Changed to from azure.servicefabric.models import FabricErrorException